sdk/daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/TailCallTest.scala

```diff
@@ -60,81 +61,93 @@ class TailCallTest(majorLanguageVersion: LanguageMajorVersion)
        case (EQUAL @Int64 x 0) of
            True -> acc
          | _    -> F:generate (Cons @Int64 [x] acc) (SUB_INT64 x 1);
-
    }
   """)
 
-  val small: Option[Int] = Some(5)
-  val unbounded: Option[Int] = None
-
   // Evaluate an expression with optionally bounded env and kont stacks
-  def runExpr(e: Ast.Expr, envBound: Option[Int], kontBound: Option[Int]): SValue = {
+  def runExpr(e: Ast.Expr, gasBudget: Long, costModel: CostModel): SValue = {
     // create the machine
-    val machine = Speedy.Machine.fromPureExpr(
-      pkgs,
-      e,
-      initialEnvSize = envBound.getOrElse(512),
-      initialKontStackSize = kontBound.getOrElse(128),
+    val machine = Speedy.UpdateMachine(
+      compiledPackages = pkgs,
+      expr = pkgs.compiler.unsafeCompile(e),
+      preparationTime = Time.Timestamp.Epoch,
+      initialSeeding = InitialSeeding.NoSeed,
+      committers = Set.empty,
+      readAs = Set.empty,
+      iterationsBetweenInterruptions = Long.MaxValue,
+      packageResolution = Map.empty,
+      costModel = costModel,
+      initialGasBudget = Some(gasBudget),
+      initialEnvSize = 1,
+      initialKontStackSize = 1,
     )
+
     // maybe replace the kont-stack with a bounded version
     // run the machine
     machine.run() match {
       case SResultFinal(v) =>
-        if (
-          envBound
-            .exists(_ < machine.env.capacity) || kontBound.exists(_ < machine.kontStack.capacity)
-        )
-          throw BoundExceeded
+        println(machine.gasBudget)
```

Comment:
Maybe we should use a Scaliest note or label here (sorry, I can't immediately remember the precise term they use) to annotate when the test fails? Thinking this might be better than the `println`.